pkg/serverless/trace/propagation/extractor.go

```diff
@@ -112,6 +113,11 @@ func (e Extractor) extract(event interface{}) (*TraceContext, error) {
 		carrier, err = headersCarrier(ev.Headers)
 	case events.LambdaFunctionURLRequest:
 		carrier, err = headersCarrier(ev.Headers)
+	case events.StepFunctionPayload:
+		tc, err := createTraceContextFromStepFunctionInput(ev)
+		if err == nil {
+			return tc, nil
+		}
```

Comment:
First looking at this, I thought there was a better way to do this. But after closer inspection, there is _another_ way to do this, but I'm not so sure it's _better_.

My biggest concern is consistency. We want to defer as much work to dd-trace-go as possible. That's what all these carriers are for -- so we can pass the set of found headers to the dd-trace-go propagator for extraction.

I threw together what this could look like, here https://github.com/DataDog/datadog-agent/commit/8ed2a33f2ae479ad0b7a24e741961e46948d0581. While this version makes the code more consistent, I'm not convinced this is any better, and in fact I think it is worse because it means creating the headers map, extracting from that map, just to later recreate it. Dumb.

So, in the end, this PR is good. I'm sad we can't be more consistent about how we are extracting trace context from step functions. Ultimately though, they are different enough to get their own way of doing things.